File: daogen/projects/5253a8bd28afd2127079661b/PersonDAO.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using platform.dao;
using platform.dao.param;

namespace platform.apptools.demo
{
    public class PersonDAO : AbstractDAO
    {
        public PersonDAO()
        {
            //注释掉此行或者赋值为string.Empty，然后配置connectionString来直连数据库
            PhysicDbName = "SysDalTest";
            ServicePort = 9000;
            CredentialID = "30303";
            base.Init();
        }

        
        // None
        public IDataReader FetchAllRecords()
        {
            try
            {
                IList<IParameter> parameters = new List<IParameter>();
                
                
                string sql = "SELECT ID,Address,Name,Telephone,Age,Gender,Birth FROM Person "   ;

                //return client.Fetch(sql, parameters);
                
                return this.Fetch(sql, parameters.ToArray());
                
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        
        // None
        public IDataReader GetByID(int iD)
        {
            try
            {
                IList<IParameter> parameters = new List<IParameter>();
                
                parameters.Add(new ConcreteParameter(){
                        DbType = DbType.Int32,
                        Name = "@ID",
                        Direction = ParameterDirection.Input,
                        Index = 0,
                        IsNullable =false,
                        IsSensitive = false,
                        Size  = 50,
                        Value = iD
                    });
                
                
                string sql = "SELECT ID,Address,Name,Telephone,Age,Gender,Birth FROM Person  WHERE  ID = @ID "   ;

                //return client.Fetch(sql, parameters);
                
                return this.Fetch(sql, parameters.ToArray());
                
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        

        
    }
}

File: dal-dao-gen/2/java/PersonDAO.java
package com.ctrip.platform.hello;

import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.List;

import com.ctrip.platform.dao.AbstractDAO;
import com.ctrip.platform.dao.enums.DbType;
import com.ctrip.platform.dao.enums.ParameterDirection;
import com.ctrip.platform.dao.param.StatementParameter;

public class PersonDAO extends AbstractDAO {

	public PersonDAO() {
		logicDbName = "SysDalTest";
		servicePort = 9000;
		credentialId = "30303";
		super.init();
	}

	public ResultSet getHello() {
		List<StatementParameter> parameters = new ArrayList<StatementParameter>();
		return this.fetch("SELECT ID,Address,Name,Telephone,Age,Gender,Birth FROM Person WHERE ", parameters, null);
	}

	
	
}
